app/controllers/subscribers_govuk_account_controller.rb

```diff
@@ -1,29 +1,41 @@
 class SubscribersGovukAccountController < ApplicationController
   before_action :validate_params
 
-  def authenticate
-    account_response = GdsApi.account_api.get_attributes(
-      attributes: %i[email email_verified],
+  before_action do
+    account_response = GdsApi.account_api.get_user(
       govuk_account_session: expected_params[:govuk_account_session],
     )
 
-    email_verified = account_response.dig("values", "email_verified")
-    email = account_response.dig("values", "email")
+    @govuk_account_id = account_response["id"]
+    email = account_response["email"]
+    email_verified = account_response["email_verified"]
 
-    api_response = { govuk_account_session: account_response["govuk_account_session"] }.compact
+    @api_response = { govuk_account_session: account_response["govuk_account_session"] }.compact
 
-    render status: :forbidden, json: api_response and return unless email_verified
-    render status: :forbidden, json: api_response and return unless email
+    if !email
+      render status: :forbidden, json: @api_response
+    elsif !email_verified
+      render status: :forbidden, json: @api_response
+    else
+      @subscriber = Subscriber.resilient_find_or_create(email, signon_user_uid: current_user.uid)
```

Comment:
Where are thse consumed as instance ariables?
I can see L27 and L32, but they seem to be taking regular variables?

And i can't see an attr_reader nywhere?

Reply:
There's an `attr_reader` line after the `private`